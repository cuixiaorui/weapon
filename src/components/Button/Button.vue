<template>
  <button class="wp-button" :class="classList" @click="onClick">
    <slot></slot>
  </button>
</template>

<script>
export default {
  props: {
    type: {
      type: String,
      default: 'default'
    },
    round: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    classList() {
      return {
        [`wp-button--${this.type}`]: true,
        'is-round': this.round
      };
    }
  },
  methods: {
    onClick() {
      this.$emit('click');
    }
  }
};
</script>

<style lang="scss" scoped>
.wp-button {
  padding: 12px 20px;
  outline: none;
  cursor: pointer;

  &--default {
    background: #fff;
    border: 1px solid #dcdfe6;
  }

  &--error {
    background-color: #f56c6c;
    border-color: #f56c6c;
  }

  &--success {
    background-color: #67c23a;
    border-color: #67c23a;
  }
}
.is-round {
  border-radius: 20px;
}
</style>
