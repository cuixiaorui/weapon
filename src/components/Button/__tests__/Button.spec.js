import Button from "../Button.vue";
import { shallowMount } from "@vue/test-utils";
const classNamePrefix = "wp-button";
describe("Button", () => {
  it("应该有类名为 wp-button 得 div", () => {
    const wrapper = shallowMount(Button);
    const result = wrapper.contains(".wp-button");
    expect(result).toBe(true);
  });

  it("可以插入文字", () => {
    const wrapper = shallowMount(Button, {
      slots: {
        default: "默认按钮"
      }
    });

    expect(wrapper.text()).toBe("默认按钮");
  });

  describe("props", () => {
    describe("type 控制按钮得类型", () => {
      const expectButtonType = function(type) {
        const wrapper = shallowMount(Button, {
          propsData: {
            type
          }
        });

        const className = `${classNamePrefix}--${type}`;
        expect(wrapper.classes(className)).toBe(true);
      };

      it("default", () => {
        const wrapper = shallowMount(Button, {});
        const className = `${classNamePrefix}--default`;
        expect(wrapper.classes(className)).toBe(true);
      });

      it("success", () => {
        expectButtonType("success");
      });
      it("error", () => {
        expectButtonType("error");
      });
    });

    it("控制显示是否为圆角", () => {
      const wrapper = shallowMount(Button, {
        propsData: {
          round: true
        }
      });

      const className = `is-round`;
      expect(wrapper.classes(className)).toBe(true);
    });
  });
  describe("events", () => {
    it("点击后应该 emit “click” 事件", () => {
      const wrapper = shallowMount(Button);
      wrapper.trigger("click");
      expect(wrapper.emitted("click")).toBeTruthy();
    });
  });
});
