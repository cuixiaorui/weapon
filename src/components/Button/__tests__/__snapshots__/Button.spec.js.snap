// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 快照 1`] = `<button class="wp-button wp-button--default"></button>`;
